internal/exeutils/exeutils.go

```diff
@@ -21,17 +22,23 @@ func Executables(bins []string) ([]string, error) {
 	exes := []string{}
 
 	for _, bin := range bins {
-		err := filepath.Walk(bin, func(fpath string, info os.FileInfo, err error) error {
-			// Filter out files that are not executable
-			if info == nil || info.IsDir() || !fileutils.IsExecutable(fpath) { // check if executable by anyone
-				return nil // not executable
+		if fileutils.IsDir(bin) {
+			entries, err := ioutil.ReadDir(bin)
+			if err != nil {
+				return nil, errs.Wrap(err, "Could not read directory")
+			}
+
+			for _, entry := range entries {
+				fpath := filepath.Join(bin, entry.Name())
+				if fileutils.IsExecutable(fpath) {
+					exes = append(exes, fpath)
+				}
 			}
+			continue
+		}
 
-			exes = append(exes, fpath)
-			return nil
-		})
-		if err != nil {
-			return exes, errs.Wrap(err, "Error while walking path")
+		if fileutils.IsExecutable(bin) {
```

Comment:
None of the bins themselves can be executables, the original code only looks for executables inside the bin paths.